service-node/src/main/scala/org/enterprisedlt/fabric/service/node/RestEndpoint.scala

```diff
@@ -330,6 +333,37 @@ class RestEndpoint(
                                 response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR)
                         }
 
+                    case "/admin/add-to-channel" =>
+                        for {
+                            state <- globalState.toRight("Node is not initialized yet")
+                            addToChannelRequest <- Try(Util.codec.fromJson(request.getReader, classOf[AddOrgToChannelRequest]))
+                              .toEither.left.map(_.getMessage)
+                            caCerts <- Try(addToChannelRequest.organizationCertificates.caCerts.map(Util.base64Decode).toIterable)
+                              .toEither.left.map(_.getMessage)
+                            tlsCACerts <- Try(addToChannelRequest.organizationCertificates.tlsCACerts.map(Util.base64Decode).toIterable)
+                              .toEither.left.map(_.getMessage)
+                            adminCerts <- Try(addToChannelRequest.organizationCertificates.adminCerts.map(Util.base64Decode).toIterable)
+                              .toEither.left.map(_.getMessage)
+                            joinResult = {
+                                logger.info(s"Adding org to channel ${addToChannelRequest.mspId} ...")
+                                state.networkManager.joinToChannel(
+                                    addToChannelRequest.channelName,
+                                    addToChannelRequest.mspId,
+                                    caCerts,
+                                    tlsCACerts,
+                                    adminCerts)
+                            }
+                        } yield joinResult match {
```

Comment:
this match will apply to joinResult, but must be applied to result of **for** expression